### ISSUE TITLE
Flutter wrap with widget, import and some other features doesn't work when working on a package

### ISSUE THREAD

**Issue by IsmailAlamKhan** [Dart-Code/Dart-Code#3325]
**Describe the bug**
I am working on a local package that I use on my projects. But after the update of dart code 3.22 can't use **wrap with widget nor import a class/ method or anything else, even can't import other packages, also the convert to expression body or block body for any method doesn't work**. I have to manually do all these. I thought it was a problem with my project so I made a new project with the package template copied the lib folder from the previous project added the dependencies to pubspec.yaml manually from the previous project it worked but when I started working on it the next day those problems appeared again. This happens to my two local package projects. Here is one of the projects <https://github.com/IsmailAlamKhan/ismails_utils>

**Versions (please complete the following information):**
 - VS Code version: 1.56.0
 - Dart extension version: 3.22.0
 - Dart/Flutter SDK version: Flutter stable channel 2.6.0 and the dev channel 2.3.0-0.1.pre.

**Comment by DanTup** [comment]
I tried with your project, but these things still seem to show up and work correctly for me:

<img width="542" alt="Screenshot 2021-05-07 at 15 06 28" src="https://user-images.githubusercontent.com/1078012/117461776-da990e00-af45-11eb-9376-340154d880fa.png">

Can you give some specific examples of which files/locations you're trying to use these in, and also what you mean by not being able to use them - are they missing from the menu, do they appear but not do anything, or do they show an error message etc.?

Thanks!

**Comment by IsmailAlamKhan** [comment]
The issue is all over the project. And sorry for not giving the info, they don't appear on the menu at all.

**Comment by DanTup** [comment]
Can you give a very specific place where you're trying and not seeing this, so I can be sure I'm looking at the same place? (or can you confirm you don't see them at the location in the screenshot above)?

Could you also confirm when you hover over the Dart/Flutter SDK version in the status bar whether you see "DAS" or "LSP" in the tooltip? Thanks!

**Comment by IsmailAlamKhan** [comment]
Yes happens for that too. And it's DAS. Also, I have two users on my PC, and the problem mostly occurs on the user that is used the most. Happens on the other too, but it's rare. And thanks for the fast reply

**Comment by DanTup** [comment]
Understood. Could you try:

1. Set the `dart.previewLsp` setting to `true` in your VS Code user settings and reload VS Code (it should now show LSP int the status bar tooltip) and confirm it still occurs
2. Using the _exact_ clone from the repo above (so I can use the same one), capture a log by:
    - Open the project and wait for analysis to complete
    - Navigate to a location where you can repro this, but put the caret on another line
    - Run the **Dart: Capture Analysis Logs** command from the command palette
    - Move the caret to the location where you want to apply the refactor and press the shortcut to open the menu
    - Click **Cancel** on the logging notification to stop logging and open the log file
   
You can attach the log here, in a gist, or email to me at logs@dartcode.org.

Thanks!

**Comment by DanTup** [comment]
@IsmailAlamKhan LSP is enabled by default for _some_ users right now (it's being ramped up progressively rather than enabled all at once). It still feels like a bug that it didn't work for non-LSP (it only really changes the protocol for communication - I didn't expect it to affect the result), although since everyone will be on LSP soon enough, it's probably not worth spending too much time on.

No problem, glad the issue is resolved! :-)

**Comment by IsmailAlamKhan** [comment]
ok thanks again.

**Comment by IsmailAlamKhan** [comment]
[Dart-Code-Log-2021-04-04 03-49-15.txt](https://github.com/Dart-Code/Dart-Code/files/6469255/Dart-Code-Log-2021-04-04.03-49-15.txt)
@DanTup Here are the logs

**Comment by DanTup** [comment]
@IsmailAlamKhan it looks like the code used in that log is different to what's currently in the repo - are you able to provide a log along with an exact commit from the repo so I can match up the locations in the code? Thanks!

**Comment by shariful2011** [comment]
@DanTup the code I used is committed to the repo, you can pull it. Or tell me which commit you have I will download that commit and send the log again. Thanks

**Comment by IsmailAlamKhan** [comment]
@DanTup the code I used is committed to the repo, you can pull it. Or tell me which commit you have I will download that commit and send the log again. Thanks

**Comment by DanTup** [comment]
@IsmailAlamKhan I cloned the latest master of that repo. The log shows code actions requested at `carousel/body.dart line 195 character 23` but there doesn't seem to be a character 23 so I can't tell exactly where that was. Thanks!

**Comment by IsmailAlamKhan** [comment]
@DanTup that is weird I will make a new log and send it now. Thanks.

**Comment by IsmailAlamKhan** [comment]
[Dart-Code-Log-2021-04-04 19-29-52.txt](https://github.com/Dart-Code/Dart-Code/files/6472717/Dart-Code-Log-2021-04-04.19-29-52.txt)
@DanTup Here you go sorry for that mistake.

**Comment by DanTup** [comment]
@IsmailAlamKhan perfect, thank you! I was able to reproduce it on stable:

<img width="351" alt="Screenshot 2021-05-13 at 14 54 14" src="https://user-images.githubusercontent.com/1078012/118135908-71a90e80-b3fb-11eb-841b-c06ff29f8dc6.png">

I tested on Flutter beta and master channels, and all of the options appear:

<img width="382" alt="Screenshot 2021-05-13 at 14 56 30" src="https://user-images.githubusercontent.com/1078012/118135936-7968b300-b3fb-11eb-9e88-870c32642a73.png">

So it seems like it's been fixed since the last Flutter stable release and should be included in the next one. Thanks!
